### PR TITLE
New routes

### PR DESCRIPTION
**Pull request by aljesusg**
So new routes for search, serializer of spin and specs

### CODE REVIEW

File: app/controllers/v1/spins_controller.rb [hunk 1]
```diff
@@ -11,18 +11,54 @@ class SpinsController < ApplicationController
...
-      render json: { data: @spins }, status: :ok
+      if params[:user_id]
+        @user = User.find_by_github_login(params[:user_id])
+        unless @user
+          return_response status: :no_content
+          return
+        end
+        @spins = Spin.where(user_id: @user.id ) if @user
+      else
+        @spins = Spin.all
```
Comment (chargio) [inline app/controllers/v1/spins_controller.rb]: It is OK for now. Opening a travis use case to make sure that in the future we only provide 10 results. We don't want to return 100's

File: app/controllers/v1/spins_controller.rb [hunk 2]
```diff
@@ -11,18 +11,54 @@ class SpinsController < ApplicationController
...
+      end
+      @spins = @spins.where('name like? or name like?', "%#{params[:query]}%", "%#{params[:query].downcase}%") if params[:query]
```
Comment (chargio) [inline app/controllers/v1/spins_controller.rb]: conditions: [ "lower(name) = ?", name.downcase ]

File: app/controllers/v1/spins_controller.rb [hunk 3]
```diff
@@ -11,18 +11,54 @@ class SpinsController < ApplicationController
...
     # Provides a view of the spin
     # TODO: When authenticated, provide extended info
+    # users/<user_id>/spins/<spin_id_or_name> Get a specific spin of user
     def show
-      @spin = Spin.find_by(id: params[:id])
-      render json: { data: @spin }, status: :ok
+      if params[:user_id]
+        @user = User.find_by_github_login(params[:user_id])
+        return_response status: :not_found unless @user
+        @spin = Spin.where(user_id: @user.id)
```
Comment (chargio) [inline app/controllers/v1/spins_controller.rb]: show shows only one. So it should be 
```
@spin = Spin.where("user_id = ? AND (id = ? OR name = ?)", 7500590, 11097119, "galaxy_demo_repos")
```

File: app/controllers/v1/spins_controller.rb (hunk 1 above)
Comment (aljesusg) [reply app/controllers/v1/spins_controller.rb]: pagination feature....

File: app/controllers/v1/spins_controller.rb (hunk 2 above)
Comment (aljesusg) [reply app/controllers/v1/spins_controller.rb]: mmmm I didn't know that we can use all psql functions in this kind of clausules in rails

**Pull request merged**